modules/c++/re/source/PCRE_STL.cpp

```diff
@@ -0,0 +1,219 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "re/PCRE.h" // this has to come before the #ifdef checks below
+
+// we use this file if we're not using actual PCRE itself
+#if defined(USE_PCRE) && defined(__CODA_CPP11)
+
+re::PCRE::PCRE(const std::string& pattern, int flags) :
+    mPattern(pattern)
+{
+    if (!mPattern.empty())
+        compile(mPattern, flags);
+}
+
+void re::PCRE::destroy()
+{
+}
+
+re::PCRE::~PCRE()
+{
+    destroy();
+}
+
+re::PCRE::PCRE(const re::PCRE& rhs)
+{
+    mPattern = rhs.mPattern;
+    compile(mPattern);
+}
+
+re::PCRE& re::PCRE::operator=(const re::PCRE& rhs)
+{
+    if (this != &rhs)
+    {
+        destroy();
+
+        mPattern = rhs.mPattern;
+
+        compile(mPattern);
+    }
+
+    return *this;
+}
+
+re::PCRE& re::PCRE::compile(const std::string& pattern,
+                            int flags)
+{
+    mPattern = (flags==PCRE_DOTALL) ? replace_dot(pattern) : pattern;
+    try 
+    {
+        mRegex = std::regex(mPattern);
+    }
+    catch (const std::regex_error& e)
+    {
+        throw PCREException(Ctxt(FmtX("PCRE std::regex constructor error: %s",
+                                      e.what())));
+    }
+
+    return *this;
+}
+
+
+const std::string& re::PCRE::getPattern() const
+{
+    return mPattern;
+}
+
+bool re::PCRE::matches(const std::string& str, int) const
+{
+    return std::regex_match(str, mRegex);
+}
+
+bool re::PCRE::match(const std::string& str,
+                     PCREMatch & matchObject,
+                     int )
+{
+    std::smatch matches;
+    bool result = std::regex_search(str, matches, mRegex);
+
+    // copy resulting substrings into matchObject
+    matchObject.resize( matches.size() );
+
+    // This causes a crash for some reason
+    //std::copy(matches.begin(), matches.end(), matchObject.begin());
+
+    for(size_t i=0; i<matches.size(); ++i) {
+        matchObject[i] = matches[i].str();
+    }
+
+    return result;
+}
+
+std::string re::PCRE::search(const std::string& matchString,
+                             int startIndex,
+                             int)
+{
+    std::smatch matches;
+
+    // search the string starting at index "startIndex"
+    bool result = std::regex_search(matchString.begin()+startIndex, 
+                                    matchString.end(), matches, mRegex);
+    
+    // if successful, return the substring matching the regex,
+    // otherwise return empty string
+    if (result && matches.size() > 0)
+    {
+        return matches[0].str();
+    }
+    else
+    {
+        return std::string("");
```

Comment:
I made the change, but doesn't that just implicitly call the std::string constructor?